File: js/securing-public-access/GetArticle.js
// DocSection: securing_public_access_get_article
// Tip: Find more about JS/TS SDKs at https://docs.kontent.ai/javascript
const KontentDelivery = require('@kentico/kontent-delivery');

const deliveryClient = new KontentDelivery.DeliveryClient({
  projectId: '<YOUR_PROJECT_ID>',
  globalQueryConfig: {
    useSecuredMode: true, // Queries the Delivery API using secure access.
  },
  secureApiKey: '<YOUR_API_KEY>',
  typeResolvers: [
    // Create strongly typed models according to https://docs.kontent.ai/strongly-typed-models
    new KontentDelivery.TypeResolver('article', (rawData) => new Article)
  ]
});

deliveryClient.item('my_article')
  .toObservable()
  .subscribe(response => console.log(response));
// EndDocSection

File: ios/securing-public-access/GetArticle.swift
// DocSection: securing_public_access_get_article
// Tip: Find more about Swift SDK at https://docs.kontent.ai/ios
import KenticoKontentDelivery

let client = DeliveryClient.init(projectId: "<YOUR_PROJECT_ID>", secureApiKey: "<YOUR_API_KEY>")

// Create strongly typed models according to https://docs.kontent.ai/strongly-typed-models
client.getItem(modelType: Article.self, itemName: "my_article") { (isSuccess, deliveryItem, error) in
     if isSuccess {
        if let article = deliveryItem.item {
            // Use your item here
        }
    } else {
        if let error = error {
            print(error)
        }
    }
// EndDocSection